### PR TITLE
Add Ruff linter. Fix formatting

### PR DESCRIPTION
**Pull request by vladsavelyev**
* Apply [ruff linter](https://github.com/ewels/MultiQC/issues/2149) to the entire code base: 
    * `ruff format .`
    * `ruff --fix .`
    * Manually addressing issues reported by `ruff .`
* Remove `OrderedDict`s (built in `dict`s are de facto sorted since Python 3.6, and guaranteed to be sorted since Python 3.7)
* Remove Python 2 code

Fixes https://github.com/ewels/MultiQC/issues/2149

### CODE REVIEW

**Comment by ewels** [comment]
Comparing the `multiqc_data` directories before and after this PR, it looks almost the same, but there are a few random things: (running with `-k yaml`).

* `multiqc_picard_insertSize.yaml` - suddenly everything goes into super deeply nested arrays, and starts talking about OrderedDicts

```diff
-14-CL_1_marked_duplicates_FR:
-  LIBRARY: ''
-  MAX_INSERT_SIZE: 243082321.0
-  MEAN_INSERT_SIZE: 171.013338
-  MEDIAN_ABSOLUTE_DEVIATION: 25.0
+ 14-CL_1_marked_duplicates_FR: !!python/object/apply:collections.OrderedDict
+ - - - SAMPLE_NAME
+    - 14-CL_1_marked_duplicates
+  - - MEDIAN_INSERT_SIZE
+    - 163.0
+  - - MEDIAN_ABSOLUTE_DEVIATION
+    - 25.0
```

* `multiqc_ivar_variants-plot.yaml` - sample `'4056145068'` vanishes

* The custom content heatmaps don't seem to work? Those results don't get included in the report.

I think that's all the meaningful changes that I can find...

**Comment by vladsavelyev** [comment]
> multiqc_picard_insertSize.yaml

Ah, didn't resolve conflicts properly when merging the sentieon-picard branch. Fixed.

The rest - '4056145068' in multiqc_ivar_variants-plot.yaml and heatmaps - has to do with a bug in custom content 🤦  Fixing.

File: multiqc/utils/report.py [hunk 1]
```diff
@@ -13,27 +13,21 @@
...
 from . import config
 
 logger = config.logger
 
-# Treat defaultdict and OrderedDict as normal dicts for YAML output
-from yaml.representer import Representer, SafeRepresenter
-
+# Treat defaultdict as normal dict for YAML output
 yaml.add_representer(defaultdict, Representer.represent_dict)
-yaml.add_representer(OrderedDict, Representer.represent_dict)
```
Comment (ewels) [inline multiqc/utils/report.py:32]: Let's keep this one please

**Review by ewels — APPROVED** [review]
Let's do it! 💪🏻

**Pull request merged**